sdk/netapp/Azure.ResourceManager.NetApp/src/Custom/Models/NetAppVolumeBackupConfiguration.Serialization.cs

```diff
@@ -0,0 +1,95 @@
+                writer.WriteBooleanValue(IsPolicyEnforced.Value);
+            }
+            if (Optional.IsDefined(VaultId))
+            {
+                writer.WritePropertyName("vaultId");
+                writer.WriteStringValue(VaultId);
+            }
+            if (Optional.IsDefined(IsBackupEnabled))
+            {
+                writer.WritePropertyName("backupEnabled");
+                writer.WriteBooleanValue(IsBackupEnabled.Value);
+            }
+            writer.WriteEndObject();
+        }
+
+        internal static NetAppVolumeBackupConfiguration DeserializeNetAppVolumeBackupConfiguration(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return null;
+            }
+            Optional<ResourceIdentifier> backupPolicyId = default;
+            Optional<bool> policyEnforced = default;
+            Optional<bool> backupEnabled = default;
+            Optional<ResourceIdentifier> vaultId = default;
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("backupPolicyId"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    backupPolicyId = new ResourceIdentifier(property.Value.GetString());
+                    continue;
+                }
+                if (property.NameEquals("policyEnforced"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    policyEnforced = property.Value.GetBoolean();
+                    continue;
+                }
+                if (property.NameEquals("backupEnabled"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    backupEnabled = property.Value.GetBoolean();
+                    continue;
+                }
+                if (property.NameEquals("vaultId"u8))
+                {
+                    if (property.Value.ValueKind == JsonValueKind.Null)
+                    {
+                        property.ThrowNonNullablePropertyIsNull();
+                        continue;
+                    }
+                    vaultId = new ResourceIdentifier(property.Value.GetString());
+                    continue;
+                }
+            }
+            return new NetAppVolumeBackupConfiguration(backupPolicyId.Value, Optional.ToNullable(policyEnforced), Optional.ToNullable(backupEnabled)){ VaultId = vaultId };
```

Comment:
Could we add a constructor that takes the extra parameter of vaultId and use that here? It should look like:
```suggestion
return new NetAppVolumeBackupConfiguration(backupPolicyId.Value, Optional.ToNullable(policyEnforced), Optional.ToNullable(backupEnabled), vaultId.Value);
```